### ISSUE TITLE
Improve the results further

### ISSUE THREAD

**Issue by anakita** [nianticlabs/monodepth2#230]
Hi,
I trained monodepth2 using rectified stereo images as input using mono_640x192 as pretrained weights,for 5k stereo pairs and 10 epochs.I tried to evaluate it and got the following result: (at cap -80m)
![80mmono2](https://user-images.githubusercontent.com/64612239/89370768-2c82de00-d6ff-11ea-85fb-522b4e7a10ae.png)
I tried using mono_stereo_1024x320 for training the second time with more images.But did not get much improvement.How do I improve my results further? Could any one help me out with this?
Thanks

**Comment by mrharicot** [comment]
Hi,
Are you training on KITTI ? or is it a custom dataset?

**Comment by anakita** [comment]
Thanks for the reply!
It is a custom dataset, with around 7k stereo pairs.

**Comment by mrharicot** [comment]
Hi,
It is very difficult to know what might be wrong without seeing the data.
Could you share a couple training pairs?

**Comment by anakita** [comment]
These are the samples of the training images
![rgb_image_390](https://user-images.githubusercontent.com/64612239/89724745-cced4100-da24-11ea-81f5-9663bb89408c.jpg)
![rgb_image_3](https://user-images.githubusercontent.com/64612239/89724746-ceb70480-da24-11ea-88a9-9c9ffd3551a4.jpg)

**Comment by mrharicot** [comment]
Thank you, I don't see anything wrong with the data at a first glance.
The error values can only be compared on kitti ground truth, it is unclear what a good metric is on your dataset.

Could you also share the output of the network for these samples and the ground truth?
Also what resolution are you training this model at? and did you update the intrinsic matrix to match that of your camera?

**Comment by anakita** [comment]
Output of the network and ground truth is as follows:
![resizergb_image_1592 jpgl_disp](https://user-images.githubusercontent.com/64612239/89738821-46267b80-da99-11ea-965b-91f52d71a03b.jpeg)
![resizergb_image_1592 jpgl](https://user-images.githubusercontent.com/64612239/89738824-47f03f00-da99-11ea-85d9-2a8c53719d9e.png)
![depth_1592](https://user-images.githubusercontent.com/64612239/89738827-4a529900-da99-11ea-8f9d-17fd27379802.jpg)
another example :
![rgb_image_394 jpg_disp](https://user-images.githubusercontent.com/64612239/89738856-7bcb6480-da99-11ea-992d-77e0e523c874.jpeg)
original image and output
![rgb_image_394 jpg](https://user-images.githubusercontent.com/64612239/89738857-7d952800-da99-11ea-9196-b69bb5fa70b9.jpeg)
original image and depth map (ground truth)

The above shown ground truth is taken from zed camera and is not reliable after a few meters,but very accurate within 15m.But when I measured the distances manually from the predicted depth maps (from the model) using depth=((baseline*focal length)/disparity)*scale factor , I did not get good accuracy.





> Also what resolution are you training this model at? and did you update the intrinsic matrix to match that of your camera?

I trained it at 1024x320 and I updated the intrinsic matrix according to the camera.

**Comment by mrharicot** [comment]
Hi,
These output depthmaps look pretty good to me!
You can't compare the numbers in kitti and the numbers on your dataset unfortunately as the statistics are quite different.
monodepth2 predicts (scaled) inverse depth, the name "disp" is actually confusing and doesn't represent a stereo disparity and I'm sorry for that.
If you know the baseline of your zed camera you can simply find the scaling factor as follows:
```python
STEREO_SCALE_FACTOR = actual_stereo_baseline_in_meters / 0.1
```
Moreover while you can initialize your network with any pretrained model, you should probably train at an aspect ratio matching your zed camera (16x9 I assume?) rather than keeping the very wide kitti aspect ratio, which should improve performance further.

**Comment by anakita** [comment]
> ```python
> STEREO_SCALE_FACTOR = actual_stereo_baseline_in_meters / 0.1
> ```
So when I use a single monocular image as input during inference,my ultimate depth will be - 
depth=(1/disp)*STEREO_SCALE_FACTOR,I have taken the disp from :

https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L133

I have used the mono_stereo mode for training.

> with an aspect ratio matching your zed camera (16x9 I assume?)

Yes I will change the resolution.I also tried incorporating the feature reconstruction loss(as mentioned in the supplementary material),which slightly improved the results.

**Comment by mrharicot** [comment]
> But the measured distances are not very accurate.

Monocular depth estimation is going to be less accurate than stereo.
You can improve the results by using more and more varied training data

> So when I use a single monocular image as input during inference,my ultimate depth will be -
depth=(1/disp)*STEREO_SCALE_FACTOR,I have taken the disp from :

That's correct.

**Comment by anakita** [comment]
Thank you for the reply!Will try out the suggestions.
